### PR TITLE
Checkbox and Local storage

### PR DESCRIPTION
**Pull request by AMkodad**
Adding checkbox to list of tasks to mark done or not done, and Using local storage for retaining data on reload

### CODE REVIEW

File: src/TodoApp.ts [hunk 1]
```diff
@@ -2,11 +2,29 @@ import { LitElement, css, html } from 'lit'
...
 
+interface TodoItem {
+    text: string;
+    completed: boolean;
+  }
+
 @customElement('todo-app')
 export class TodoApp extends LitElement {
-  @property({ type: Array }) todos: string[] = [];
+  @property({ type: Array }) todos: TodoItem[] = [];
```
Comment (NeelSinha) [inline src/TodoApp.ts]: @state, learn difference between state and property

File: src/TodoApp.ts [hunk 2]
```diff
@@ -2,11 +2,29 @@ import { LitElement, css, html } from 'lit'
...
   @property({ type: String }) newTodo: string = '';
   @property({ type: Number }) deleteIconIndex: number | null = null;
+  @property({ type: Number }) pendingTasksCount: number = 0;
```
Comment (NeelSinha) [inline src/TodoApp.ts]: is a state not property

File: src/TodoApp.ts [hunk 3]
```diff
@@ -134,11 +163,32 @@ export class TodoApp extends LitElement {
 
   addTodo() {
     if (this.newTodo.trim() !== '') {
-      this.todos = [...this.todos, this.newTodo.trim()];
-      this.newTodo = '';
-    }
+        this.todos = [...this.todos, { text: this.newTodo.trim(), completed: false }];
+        this.pendingTasksCount++;
+        this.updatePendingTasksCount();
```
Comment (NeelSinha) [inline src/TodoApp.ts:149]: line 167 and 168 are doing the same thing ?

File: src/TodoApp.ts [hunk 4]
```diff
@@ -134,11 +163,32 @@ export class TodoApp extends LitElement {
...
+        this.saveTodosToLocalStorage();
+        this.newTodo = '';
+      }
   }
 
+  toggleStrikeout(index: number, checked: boolean) {
```
Comment (NeelSinha) [inline src/TodoApp.ts]: a toggle function should toggle the done / undone state, we do not need second parameter as checked here

File: src/TodoApp.ts [hunk 5]
```diff
@@ -148,14 +198,29 @@ export class TodoApp extends LitElement {
...
   }
 
   clearAll() {
     this.todos = [];
+    this.pendingTasksCount = 0; 
+    this.saveTodosToLocalStorage();
     this.requestUpdate();
   }
+
+  updatePendingTasksCount() {
```
Comment (NeelSinha) [inline src/TodoApp.ts]: all these functions are used before they are declared, fix that

**Review by NeelSinha — COMMENTED** [review]
comments

File: src/TodoApp.ts (hunk 1 above)
Comment (AMkodad) [reply src/TodoApp.ts]: **Done**

File: src/TodoApp.ts (hunk 2 above)
Comment (AMkodad) [reply src/TodoApp.ts]: **Done**

File: src/TodoApp.ts (hunk 3 above)
Comment (AMkodad) [reply src/TodoApp.ts:149]: **Yeah Did the necessary changes**

File: src/TodoApp.ts (hunk 4 above)
Comment (AMkodad) [reply src/TodoApp.ts]: **Yeah Done**

File: src/TodoApp.ts (hunk 5 above)
Comment (AMkodad) [reply src/TodoApp.ts]: **Yeah done**

**Pull request merged**